### ISSUE TITLE
[FEATURE]: when train_dreambooth_colossalai support train_text_encoder?

### ISSUE THREAD

**Issue by vonchenplus** [hpcaitech/ColossalAI#2309]
### Describe the feature

I found that the train_dreambooth_colossalai script you provide no longer supports train_text_encoder parameter, 
but it's gives much better results especially on faces, 
so I would like to ask if you have plans to support this parameter?

Thanks again!

**Comment by blx0102** [comment]
+1

**Comment by mingqizhang** [comment]
Have you solve this problem?

**Comment by vonchenplus** [comment]
No, ColossalAI's optimizer cannot support multiple network parameters.

**Comment by feifeibear** [comment]
You are right. The GeminiDPP can only deal with parameters from a single model, rather than parameters from multiple models. But, you can support train_text_encoder parameter with some code modifications.

**Comment by feifeibear** [comment]
We are considering making Gemini more flexible, but we have no such plan in the next two weeks.

**Comment by haofanwang** [comment]
+1. It would be much more flexible if we can train multi-modules, which is common when finetuning stable diffusion.

**Comment by haofanwang** [comment]
@feifeibear any update for this?

**Comment by hdjsjyl** [comment]
hi @feifeibear , 
thanks for your great job! 
Would you explain a little bit about which part I should change for supporting text_encoder parameters?
Any help would be appreciated! Thank you so much!

**Comment by feifeibear** [comment]
We have a very intensive discussion on this.
Can could please report the latest conclusion @1SAA , @ver217 .

**Comment by shileims** [comment]
Hi @feifeibear ,
     Thanks for your reply.

Hi @1SAA  and @ver217 ,
     would you share your latest conclusion here? 
    I am looking forward to your reply.

Thank you so much!

**Comment by shileims** [comment]
Hi @feifeibear , @1SAA ， and @ver217 ,
     Any advice would be very appreciated since I am stuck here for a long time.
     Really looking forward to your reply. Pls feel free to update your solution here, I can validate the idea. Thank you so much!

**Comment by shileims** [comment]
Hi @feifeibear , @1SAA , @ver217 ,
     Any chance to get your discussion here? It will be very helpful. Looking forward to your reply. Thank you so much!

**Comment by shileims** [comment]
Hi @feifeibear , @1SAA , @ver217 ,
    Could you spend a little bit of time describing your discussion here? 
    Because it is beneficial for many users and in many cases. 
    Thank you so much!

**Comment by shileims** [comment]
Hi @feifeibear , @1SAA , @ver217 ,
     Hi authors of Colossalai, would you provide some useful suggestions for training text encoder and unet together? Thanks

**Comment by shileims** [comment]
Hi @feifeibear , @1SAA , @ver217 ,
     Any updates for this question? I am still waiting for your answers.
     Any answer will be appreciated! Thanks

**Comment by ver217** [comment]
@Fazziekey Could you answer this question?
